udata/core/dataset/rdf.py

```diff
@@ -614,42 +620,44 @@ def dataset_from_rdf(graph: Graph, dataset=None, node=None):
         dataset.temporal_coverage = temporal_from_rdf(d.value(DCT.temporal))
 
     # Adding some metadata to extras - may be moved to property if relevant
-    provenance = [p.value(RDFS.label) for p in d.objects(DCT.provenance)]
+    provenance = rdf_values(d, DCT.provenance, parse_label=True)
     if provenance:
-        dataset.extras["harvest"] = {
-            "dct:provenance": provenance,
-            **dataset.extras.get("harvest", {}),
-        }
+        add_dcat_extra(dataset, "provenance", provenance)
 
-    resources_licenses = set()
+    resources_licenses_hints = set()
+    resources_access_rights = []
     for distrib in d.objects(DCAT.distribution | DCAT.distributions):
         resource_from_rdf(distrib, dataset)
-        resources_licenses |= licenses_from_rdf(distrib)
+        resources_access_rights.append(access_rights_from_rdf(distrib))
+        # include both dct:license and dct:rights as licenses hints from resources
+        resources_licenses_hints |= licenses_from_rdf(distrib)
+        resources_licenses_hints |= rights_from_rdf(distrib)
 
-    # assign the distribution accessRights to the dataset if all distribs that have an accessRights have the same accessRights
-    # and the dataset doesn't have accessRights
+    # assign the common resources accessRights to the dataset if it doesn't have accessRights
     dataset_access_rights = access_rights_from_rdf(d)
-    if not dataset_access_rights:
-        distribs_access_rights = set(
-            resource.extras["harvest"]["dct:accessRights"]
-            for resource in dataset.resources
-            if resource.extras.get("harvest", {}).get("dct:accessRights") is not None
-        )
-        if len(distribs_access_rights) == 1:
-            dataset_access_rights = distribs_access_rights.pop()
-
+    if not dataset_access_rights and resources_access_rights:
+        dataset_access_rights = set.intersection(*resources_access_rights)
```

Comment:
This doesn't enforce that ALL resources have the value, only that it's unique amongst the resources that do have a value.

We previously relaxed that constraint but my latest understanding is that we might want to reintroduce it. That way:
- If the original metadata was directly encoded in DCAT, we only move resource-level constraints up when all resources are concerned.
- Id the original metadata was ISO converted through SEMIC, we're guaranteed by design that all resources will have the exact same constraints.

Reply:
```
>>> set.intersection({1,2}, {1,3}, {})
set()
```

Should work as we want?

Reply:
Indeed, I expected `resources_access_rights` to only contain entries with values, but since it also contains empty sets, this is OK.

Might be worth adding a comment/assert about that assumption? If we were to change `resources_access_rights.append(access_rights_from_rdf(distrib))` above to skip empty entries (as I might have done out of habit, hence my erroneous comment), it would keep working but on a different logic.

Reply:
> we may want to move constraints at the dataset level only if all resources have all the exact same constraints?

Works for us too

Reply:
Changed it to "exactly the same unique set for all resources". Very sad I can't use `set.intersection` anymore.

Reply:
You still could... :nail_care:
```
resources_access_rights[0] == set.intersection(*resources_access_rights)
```

Reply:
You still could... :nail_care:
```
set.union(*resources_access_rights) == set.intersection(*resources_access_rights)
```